### PR TITLE
Add getting-started example for create-react-app (#920)

### PR DESCRIPTION
**Pull request by RobJacobson**
Here's a first attempt at adding a minimal example using create-react-app.  How does this look?

### CODE REVIEW

**Comment by RobJacobson** [comment]
Hmm... I'm confused about the Travis error.  When I run the tests in my CLI, I get this error message (which isn't terribly helpful):

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! react-map-gl@5.2.0 test: `flow check && ocular-test`
npm ERR! Exit status 1

Have I made a mistake or omitted something necessary?  (This is my first pull request here.)  Thanks in advance.

File: examples/get-started/create-react-app/package.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    "react-scripts": "3.3.0"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject"
+  },
+  "eslintConfig": {
+    "extends": "react-app"
```
Comment (Pessimistress) [inline examples/get-started/create-react-app/package.json]: This is breaking CI. You need to either remove this or add `eslint-config-react-app` to the root `devDependencies`.

File: examples/get-started/create-react-app/src/map.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  };
+
+  render() {
+    return (
+      <MapGL
+        {...this.state.viewport}
+        width="100%"
+        height="100vh"
+        mapboxApiAccessToken={API_KEY}
+        onViewportChange={viewport => this.setState({ viewport })}
```
Comment (Pessimistress) [inline examples/get-started/create-react-app/src/map.jsx]: `this.updateViewport`

File: examples/get-started/create-react-app/src/map.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+class Map extends Component {
+  state = {
+    viewport: {
+      latitude: 37.785164,
+      longitude: -100,
+      zoom: 3.5,
+      bearing: 0,
+      pitch: 0
+    },
+    popupInfo: null
```
Comment (Pessimistress) [inline examples/get-started/create-react-app/src/map.jsx]: This doesn't seem to be used

File: examples/get-started/create-react-app/package.json (hunk 1 above)
Comment (RobJacobson) [reply examples/get-started/create-react-app/package.json]: Thanks. I added the dev dependency.

File: examples/get-started/create-react-app/src/map.jsx (hunk 2 above)
Comment (RobJacobson) [reply examples/get-started/create-react-app/src/map.jsx]: Good catch!

File: examples/get-started/create-react-app/src/map.jsx (hunk 3 above)
Comment (RobJacobson) [reply examples/get-started/create-react-app/src/map.jsx]: True.  (A copy-paste error.)

File: examples/get-started/create-react-app/package.json (hunk 1 above)
Comment (RobJacobson) [reply examples/get-started/create-react-app/package.json]: Thanks. I added eslint-config-react-app.

File: examples/get-started/create-react-app/src/map.jsx (hunk 3 above)
Comment (RobJacobson) [reply examples/get-started/create-react-app/src/map.jsx]: Good eye!  Sorry for the copy-paste error.

**Comment by RobJacobson** [comment]
I've made the requested changes.  However, I'm still failing the Travis CI check.  Can you provide any guidance?

File: examples/get-started/create-react-app/package.json (hunk 1 above)
Comment (Pessimistress) [reply examples/get-started/create-react-app/package.json]: You need to add it to the root package.json. CI does not install dependencies for each example.

File: examples/get-started/create-react-app/.gitignore [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
```
Comment (Pessimistress) [inline examples/get-started/create-react-app/.gitignore]: Do you need this file specifically for this example? Can this be rolled into the root .gitignore?

**Comment by Pessimistress** [comment]
> I've made the requested changes. However, I'm still failing the Travis CI check. 

You need to add it to the root package.json. CI does not install dependencies for each example.

**Comment by RobJacobson** [comment]
I'm terribly sorry, but the Travis build failed again.  I removed "eslint-config-react-app" from the example's config.json file and moved it to config.json in the root project.  Could you please provide a further hint?  (The Travis CI check seems like a "black box" to me.)

**Comment by Pessimistress** [comment]
> I removed "eslint-config-react-app" from the example's config.json file and moved it to config.json in the root project.

CI runs `yarn` at root. If `eslint-config-react-app` is not a root dependency, then it won't be installed.

If you don't really need the eslint setup for your example, then just remove it period. The repo has its own eslint configuration.

**Comment by Pessimistress** [comment]
Looks like you have committed some irrelevant files in `examples/geojson-markers`.

**Comment by RobJacobson** [comment]
Thank you.  I've done the following:
1. Deleted esLintConfig from package.json for the CRA example.
2. Deleted eslint-config-react-app from root/package.json.
3. Deleted the extraneous files.  (Sorry about that.)

Unfortunately, the Travis check is still failing.  I'm feeling like an idiot.  Could you please elaborate again on the problem?

**Comment by Pessimistress** [comment]
CI is failing for a different issue (lint error). 

```
/home/travis/build/uber/react-map-gl/examples/get-started/create-react-app/src/index.js
  5:26  error  'document' is not defined  no-undef
```

Have you run `yarn` at root before you commit?

**Comment by RobJacobson** [comment]
Thanks for your tip.  I'm using VS Code with the ESLint extension.  I'm getting the following error message at the top level of the project:

```
ESLint: ESLint configuration in .eslintrc.js is invalid: - Property "overrides" is the wrong type (expected array but got `{"files":["*.spec.js","webpack.config.js","**/bundle/*.js"],"rules":{"import/no-extraneous-dependencies":0}}`). . Please see the 'ESLint' output channel for details.
```

What am I doing wrong?  Thanks for your patience with this issue.

**Comment by Pessimistress** [comment]
This issue is recently fixed, you may need to rebase master.

**Pull request closed without merging**